site/tests/test_components.py

```diff
@@ -71,9 +78,267 @@ def test_subjects_validation_component_update_draft_admin(
+    )
+
+    # Verify only one PID remains (deletion worked)
+    pids_after_deletion = PersistentIdentifier.query.filter(
+        PersistentIdentifier.object_uuid == record7.pid.object_uuid,
+        PersistentIdentifier.pid_type != record7.pid.pid_type,
+    ).all()
+    assert len(pids_after_deletion) == 1
+    assert pids_after_deletion[0].pid_type == "cdsrn"
+    assert pids_after_deletion[0].pid_value == "1234567895"
+
+    # 6. Test PID creation: Add a new identifier
+    draft7.metadata["identifiers"] = [
+        {"scheme": "cdsrn", "identifier": "1234567895"},  # Keep existing
+        {"scheme": "testrn", "identifier": "1234567897"},  # Add new
+    ]
+    assert (
+        component.publish(administrator.identity, draft=draft7, record=record7) == None
+    )
+
+    # Verify two PIDs exist now (creation worked)
+    pids_after_creation = PersistentIdentifier.query.filter(
+        PersistentIdentifier.object_uuid == record7.pid.object_uuid,
+        PersistentIdentifier.pid_type != record7.pid.pid_type,
+    ).all()
+    assert len(pids_after_creation) == 2
+    pid_types_after_creation = {pid.pid_type for pid in pids_after_creation}
+    assert pid_types_after_creation == {"cdsrn", "testrn"}
+
+    # 6. Test PID updates: Change an identifier value
+    draft7.metadata["identifiers"] = [
+        {"scheme": "cdsrn", "identifier": "1234567898"},  # Changed value
+        {"scheme": "testrn", "identifier": "1234567897"},  # Keep this one
+    ]
+    assert (
+        component.publish(administrator.identity, draft=draft7, record=record7) == None
+    )
+
+    # Verify old PID was deleted and new one was created (update worked)
+    pids_after_update = PersistentIdentifier.query.filter(
+        PersistentIdentifier.object_uuid == record7.pid.object_uuid,
+        PersistentIdentifier.pid_type != record7.pid.pid_type,
+    ).all()
+    assert len(pids_after_update) == 2
+    assert {pid.pid_value for pid in pids_after_update} == {"1234567898", "1234567897"}
+
+    # Verify old PID no longer exists
+    old_pids = PersistentIdentifier.query.filter(
+        PersistentIdentifier.pid_type == "cdsrn",
+        PersistentIdentifier.pid_value == "1234567895",
+    ).all()
+    assert len(old_pids) == 0
+
+    # 7. Duplicate validation
+    draft9 = service.create(uploader.identity, minimal_restricted_record)
+    draft9._record.metadata["identifiers"] = [
+        {
+            "scheme": "cdsrn",
+            "identifier": "1234567890",
+        },  # This already exists from draft1
+    ]
+
+    errors = []
+    component.update_draft(
+        administrator.identity, data=draft9.data, record=draft9._record, errors=errors
+    )
+    assert len(errors) > 0
+    # This should raise PIDAlreadyExists error
+    record9 = RDMRecord.publish(draft9._record)
+    with pytest.raises(PIDAlreadyExists):
+        component.publish(administrator.identity, draft=draft9._record, record=record9)
```

Comment:
Thanks for adding all these tests, super good.
Can you please only change the identity, and always use the normal user and not system or admin whenever possible?